crates/wasm-mutate/src/mutators/peephole/eggsy/encoder/mod.rs

```diff
@@ -27,7 +29,7 @@ impl Encoder {
     /// Reassembles the mutated function and return a `Function` entry
     pub fn build_function(
         info: &ModuleInfo,
-        rnd: &mut rand::prelude::SmallRng,
+        rnd: &RefCell<&mut rand::prelude::SmallRng>,
```

Comment:
Wrapping this in a `RefCell` doesn't seem like it should be necessary, can you explain why we need it here?

Reply:
It is passed through several closures, I think this prevents borrowing issues

Reply:
Rather than having the closures capture the RNG, it could be a parameter passed to each of them. This should remove the need for the `RefCell`

Reply:
Or is the issue that the closures that capture the RNG aren't things you can add new parameters to, but instead have to conform to an existing interface, like `Iterator` methods?

Basically this error: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=e63e24bfcf10504ff8b3927803479bd9

If so, it should be possible to work around this by refactoring the sequence of `Iterator` methods that are chained into a single `filter_map` or `flat_map` call rather than a series of `filter`s and `map`s.

Reply:
What if we use `Rc` and `RefCell` to remove the references from the method signature.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=2c6ec744fd1fd8dde78334221aafd369